06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js

```diff
@@ -0,0 +1,255 @@
+   * booleans to indicate if the words are present in each book
+   * e.g [true, false, false, false, false, true . . .]
+   * @param {Integer} bookCount - The number of the book present in the file
+   * @returns {Array} - An Array that consist of sub arrays of each word
+   * and a boolean to indicate their presence in each book
+   * @memberOf InvertedIndexUtility
+   */
+  static getEachBookOccurence(checked, bookCount) {
+    const result = [];
+    let subResult = [];
+    let index = 0;
+    checked.forEach((check) => {
+      if (index < bookCount) {
+        subResult.push(check);
+        index += 1;
+      }
+
+      if (index === bookCount) {
+        result.push(subResult);
+        index = 0;
+        subResult = [];
+      }
+    });
+    return result;
+  }
+
+  /**
+   * This method returns an array that consists of sub array that consist of a
+   *  word and a boolean that indicates it's presence in each book
+   * e.g [['Alice', true, false, false], ['in', false, false, true], ... ]
+   * @param {Array} words - An array of all words in a given book text
+   * @param {Array} wordsOccurrence  An Array that consist of sub arrays of
+   * each word and a boolean to indicate their presence in each book
+   * @returns {Array} - The array consist of a sub array of each word present
+   * in each book and a boolean to indicate their presence in each boom
+   * @memberOf InvertedIndexUtility
+   */
+  static mapWordsWithOccurence(words, wordsOccurrence) {
+    let index = 0;
+    words.forEach((word) => {
+      wordsOccurrence[index].unshift(word);
+      index += 1;
+    });
+    return wordsOccurrence;
+  }
+
+  /**
+   * This method returns the index of a fileName if present in a given Array.
+   * The method returns -1 if it isn't present in the given Array.
+   * @static
+   * @param {Array} fileNames
+   * @param {String} fileName
+   * @returns {Integer} - The index of the fileName
+   * @memberOf InvertedIndexUtility
+   */
+  static getSelectedIndex(fileNames, fileName) {
+    return fileNames.indexOf(fileName);
+  }
+
+  /**
+   * This method checks if the file is valid JSON
+   * @param {JSON} file
+   * @returns {boolean} - return true or false depending
+   * on whether the file is valid JSON
+   * @memberOf Validations
+   */
+  static isValidJson(file) {
+    try {
+      JSON.parse(file);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /**
+   * This method checks if the content of a JSON file is valid
+   * @param {JSON} file - The content of the JSON file
+   * @returns {boolean} - returns true or false depending
+   * on whether the content of the JSON file is valid
+   * @memberOf InvertedIndex
+   */
+  static isValidContent(file) {
+    let isValid = true;
+    if (file.length === 0) {
+      return false;
+    }
+    file.forEach((book) => {
+      if ((!book.title) || (!book.text)) {
+        isValid = false;
+      }
+    });
+    return isValid;
+  }
+}
```

Comment:
I have another method called isValidContent which checks if the content of the JSON file is valid.